crypto/rand/rand_lib.c

```diff
@@ -1023,4 +1138,60 @@ int RAND_set_seed_source_type(OSSL_LIB_CTX *ctx, const char *seed,
         && random_set_string(&dgbl->seed_propq, propq);
 }
 
-#endif
+int RAND_set1_random_provider(OSSL_LIB_CTX *ctx, OSSL_PROVIDER *prov)
+{
+    RAND_GLOBAL *dgbl = rand_get_global(ctx);
+
+    if (dgbl == NULL)
+        return 0;
+
+    if (prov == NULL) {
+        free_random_provider_name(dgbl);
+        dgbl->random_provider_name = NULL;
+        dgbl->random_provider = NULL;
+        return 1;
+    }
+
+    if (dgbl->random_provider == prov)
+        return 1;
+
+    if (!set_random_provider_name(dgbl, OSSL_PROVIDER_get0_name(prov)))
+        return 0;
+
+    dgbl->random_provider = prov;
+    return 1;
+}
+
+int ossl_rand_check_random_provider_load(OSSL_LIB_CTX *ctx, OSSL_PROVIDER *prov)
```

Comment:
If we take out the provider load in this code then we don't  the load or unload functions.
Again - if there is a driving reason why we have to do this here it should noted explicitly why.

Reply:
Comment added.  These handle the case where the name is specified and the provider is loaded later.